packages/bpk-react-utils/src/BpkDialogWrapper/BpkDialogWrapper.tsx

```diff
@@ -0,0 +1,181 @@
+  isDialogOpen: boolean;
+};
+
+// TODO: this check if the browser support the HTML dialog element. We can remove it once we drop support as a business for Safari 14
+const dialogSupported = typeof HTMLDialogElement === 'function';
+
+const setPageProperties = ({ isDialogOpen }: DialogProps) => {
+  document.body.style.overflowY = isDialogOpen ? 'hidden' : 'visible';
+
+  if (!dialogSupported) {
+    document.body.style.position = isDialogOpen ? 'fixed' : 'relative';
+    document.body.style.width = isDialogOpen ? '100%' : 'auto';
+  }
+};
+
+export const BpkDialogWrapper = (props: Props) => {
+  const {
+    ariaLabelledby,
+    children,
+    closeOnEscPressed = false,
+    closeOnScrimClick = false,
+    dialogClassName = '',
+    exiting = false,
+    id,
+    isOpen,
+    onClose,
+    timeout = {appear: 0, exit: 0},
+    transitionClassNames = {}
+  } = props;
+
+  const ref = useRef<HTMLDialogElement>(null);
+  const [dialogTarget, setDialogTarget] = useState<HTMLElement | null>(null);
+
+  useEffect(() => {
+    const dialog = document.getElementById(`${id}`);
+    const dialogWithPolyfill = document.getElementById(`${id}-polyfill`);
+
+    const handleBackdropClick = (modal: HTMLElement | null) => {
+      if (closeOnScrimClick && modal) {
+        modal.addEventListener('click', (event: MouseEvent | PointerEvent) => {
+          const { target } = event;
+
+          if (target === modal) {
+            onClose(event, {source: "DOCUMENT_CLICK"});
+            event.stopPropagation();
+          }
+        });
+      }
+    };
+
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (closeOnEscPressed
+        && event.key === 'Escape'
+        && (!dialogWithPolyfill || event.target === dialogWithPolyfill)) {
+        onClose(event, {source: "ESCAPE"});
+      }
+      event.stopPropagation();
+    };
+
+    if (isOpen) {
+      // There is a bug on older versions of browser using chromium (chrome, firefox, edge >114) where the dialog got an open attribute even before it is opened.
+      // Therefore, when trying to open it, it crashes and log an error mentioning the dialog has already an open attribute.
+      ref.current?.removeAttribute('open');
+      ref.current?.showModal?.();
+
+      if (dialogWithPolyfill) {
+        setDialogTarget(dialogWithPolyfill);
+        handleBackdropClick(dialogWithPolyfill);
+        window.addEventListener('keydown', handleKeyDown);
+      } else {
+        setDialogTarget(dialog);
+      }
+      handleBackdropClick(dialog);
+    } else {
+      ref.current?.close?.();
+    }
+
+    setPageProperties({ isDialogOpen: isOpen });
+    return () => {
+      setPageProperties({ isDialogOpen: false })
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [id, isOpen, onClose, closeOnEscPressed, closeOnScrimClick]);
+
+  return isOpen ? (
+    <div
+      className={getClassName(
+        'bpk-dialog-wrapper',
+        dialogSupported ? '' : 'bpk-dialog-wrapper--polyfill',
```

Comment:
I believe this could be simplified to 
```suggestion
        !dialogSupported && 'bpk-dialog-wrapper--polyfill',
```

Reply:
Need empty string here, otherwise the `className` computed is incorrect

Reply:
Oops noob move on my part 😅